examples/animation/color_animation.rs

```diff
@@ -0,0 +1,173 @@
+//! Demonstrates how to animate colors in different color spaces using mixing and splines.
+
+use bevy::{
+    math::cubic_splines::{CubicCurve, Point},
+    prelude::*,
+};
+
+// We define this trait so we can reuse the same code for multiple color types that may be implemented using curves.
+trait CurveColor: Point + Into<Color> + Send + Sync + 'static {}
+impl CurveColor for LinearRgba {}
+impl CurveColor for Oklaba {}
+impl CurveColor for Xyza {}
+
+// We define this trait so we can reuse the same code for multiple color types that may be implemented using mixing.
+trait MixedColor: Mix + Into<Color> + Send + Sync + 'static {}
+impl MixedColor for Oklcha {}
+impl MixedColor for Srgba {}
+impl MixedColor for Hsla {}
+
+#[derive(Debug, Component)]
+struct Curve<T: CurveColor>(CubicCurve<T>);
+
+#[derive(Debug, Component)]
+struct Mixed<T: MixedColor>([T; 4]);
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(
+            Update,
+            (
+                animate_curve::<LinearRgba>,
+                animate_curve::<Oklaba>,
+                animate_curve::<Xyza>,
+                animate_mixed::<Hsla>,
+                animate_mixed::<Srgba>,
+                animate_mixed::<Oklcha>,
+            ),
+        )
+        .run();
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+
+    // The color spaces `Oklaba`, `Laba`, `LinearRgba` and `Xyza` all are either perceptually or physically linear.
+    // This property allows us to define curves, e.g. bezier curves through these spaces.
+
+    // Define the control points for the curve.
+    // For more information, please see the cubic curve example.
+    let points = [
+        LinearRgba::WHITE,
+        LinearRgba::rgb(1., 1., 0.), // Yellow
+        LinearRgba::RED,
+        LinearRgba::BLACK,
+    ];
+    // Spawn a sprite using the provided control points.
+    spawn_curve_sprite(&mut commands, 275., points);
+
+    let points = [
+        Xyza::xyz(1., 1., 1.),                // White
+        Xyza::xyz(0.76998, 0.92781, 0.13853), // Yellow
+        Xyza::xyz(0.41239, 0.21264, 0.01933), // Red
+        Xyza::xyz(0., 0., 0.),                // Black
```

Comment:
I think it would be nice if all the key points were defined using the same 4 colours, with the appropriate `T:from` used to get them in the correct space.